### PR TITLE
Add roadmap documentation.

### PR DESCRIPTION
**Pull request by juliusv**
@beorn7 @brian-brazil

### CODE REVIEW

File: content/docs/introduction/roadmap.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+GitHub issue: [#9](https://github.com/prometheus/prometheus/issues/9)
+
+**More flexible label matching in binary operations**
+
+[Binary operations](/docs/querying/operators/) between time series vectors
+currently require exact matches of label sets on both sides of the operation
+in order for paired vector elements to propagate into the result. We plan to
+add more flexible label-matching features to support advanced query use cases.
+
+GitHub issue: [#393](https://github.com/prometheus/prometheus/issues/393)
```
Comment (brian-brazil) [inline content/docs/introduction/roadmap.md]: That's just one of the features we'll be adding.

File: content/docs/introduction/roadmap.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+title: Roadmap
+sort_rank: 6
+---
+
+# Roadmap
```
Comment (brian-brazil) [inline content/docs/introduction/roadmap.md]: More clients, more exporters?

**Comment by beorn7** [comment]
Perhaps mention our vision of actually making use of the type of a metric to find out about semantically wrong expressions and warn the user.

**Comment by juliusv** [comment]
@beorn7 Hmm, I would have considered that more of a nice-to-have minor feature that is not firmly planned yet. Whereas the rest are quite vital stuff we really need for Prometheus to be successful in the future. But open to debate...

File: content/docs/introduction/roadmap.md (hunk 1 above)
Comment (juliusv) [inline content/docs/introduction/roadmap.md]: It is, but I found it a quite vital one. What do you propose? Renaming the section to "Query language improvements" and listing several examples?

File: content/docs/introduction/roadmap.md (hunk 2 above)
Comment (juliusv) [inline content/docs/introduction/roadmap.md]: Besides the Python client lib, do we have more concrete plans here?

File: content/docs/introduction/roadmap.md (hunk 1 above)
Comment (brian-brazil) [inline content/docs/introduction/roadmap.md]: Maybe clarify that the github issue is just one of the features planned for binary operations?

**Comment by beorn7** [comment]
@juliusv  The problem I see without that note is that smart people will ask us why we have types in the exposition format at all (and not just 'gauges'), Cf. discussion with our principal engineer. Without the vision behind it, it seems like unneeded complexity.

File: content/docs/introduction/roadmap.md (hunk 1 above)
Comment (juliusv) [inline content/docs/introduction/roadmap.md]: Ah, what else do we have planned for binops? If we do have other things, we should create issues for them.

Comment (brian-brazil) [inline content/docs/introduction/roadmap.md]: Matching on a subset of labels without the logic in that issue, and various modifiers as to what happens to the labels you don't match on

**Comment by juliusv** [comment]
Added second binop issue and section about metric types. @brian-brazil feel free to add more sections about exporters, client libraries, staleness.

**Pull request merged**